app/controllers/message_threads/bulk/signings_controller.rb

```diff
@@ -0,0 +1,48 @@
+module MessageThreads
+  module Bulk
+    class SigningsController < ::ApplicationController
+      before_action :set_message_ids
+      before_action :set_message_objects
+      before_action :nothing_to_sign_redirect, only: %i[new start]
+
+      include TurboReload
+
+      def new
+        authorize MessageObjectsTag
+      end
+
+      def start
+        authorize MessageObjectsTag
+      end
+
+      def update
+        authorize MessageObjectsTag
+
+        if params[:result] == "ok"
+          redirect_back fallback_location: message_threads_path, notice: t("signing.processed"), status: 303
+        else
+          redirect_back fallback_location: message_threads_path, alert: t("signing.failed"), status: 303
+        end
+      end
+
+      private
+
+      def set_message_ids
+        @message_thread_ids = params[:message_thread_ids] || []
+      end
+
+      def set_message_objects
+        @message_objects = MessageObject.joins(:tags, message: :thread).
+          where(message_threads: { id: @message_thread_ids }).
+          where(tags: { id: Current.user.signature_requested_from_tag })
```

Comment:
Tu nie je ziadna kontrola tenanta? Nemalo by sa ist vzdy cez `Current.tenant.message_objects...` 

Ono asi sa nic velmi nema co stat lebo je tam este aj limitacia na tag usera, ale aj tak mi to pride fragilne.

Reply:
vyriesil com cez `policy_scope`